Connection/Connection.php

```diff
@@ -650,7 +643,11 @@ public function isRemoteAddressConsidered()
      */
     public function getRemoteAddress()
     {
-        return $this->_remoteAddress;
+        if (null === $this->_node) {
+            return null;
+        }
+
+        return $this->_node->getPeerName();
```

Comment:
because the original method behind are `stream_socket_get_name` and the second argument is called `want_peer` in php doc